index.js

```diff
@@ -762,6 +765,14 @@
     typeEq('sanctuary-def/Type')
   );
 
+  var VariadicType = NullaryType(
+    'sanctuary-def/VariadicType',
+    '',
+    function(x) {
+      return test([], Type, x) || test([], AnyFunction, x);
```

Comment:
This predicate looks fine to me. I think the ultimate solution is to require values such as `$.Array($.Unknown)` and `S.EitherType($.Unknown, $.Unknown)` in place of `$.Array` and `S.EitherType`. I hope VariadicType will be short-lived, so the predicate need not be perfect.